acado/matrix_vector/block_matrix.cpp

```diff
@@ -108,6 +152,40 @@ BlockMatrix BlockMatrix::operator+( const BlockMatrix& arg ) const{
 	return tmp;
 }
 
+void BlockMatrix::mat_mat_add(const BlockMatrix& m1, const BlockMatrix& arg, BlockMatrix& result)
+{
```

Comment:
All the matrix math is a copy-paste of existing methods and tweaked for static functions and provided results matrix.